src/app_service/service/community/service.nim

```diff
@@ -1351,6 +1356,59 @@ QtObject:
       self.events.emit(SIGNAL_COMMUNITY_IMPORTED, CommunityArgs(community: community))
 
     self.events.emit(SIGNAL_COMMUNITY_DATA_IMPORTED, CommunityArgs(community: community))
+  
+  proc removeLoadingRequestToJoinCommunity(self: Service, communityId: string) =
+    var i = 0
+    for communityRequest in self.myCommunityRequests:
+      if (communityRequest.communityId == communityId and RequestToJoinType(communityRequest.state) == RequestToJoinType.Loading):
+          self.myCommunityRequests.delete(i)
+          self.events.emit(SIGNAL_REQUEST_TO_JOIN_COMMUNITY_DELETED, CommunityRequestToJoinArgs(communityId: communityId))
+          return
+      i.inc()
+
+  proc asyncRequestToJoinCommunity*(self: Service, communityId: string, ensName: string, password: string) =
+    try:
+      let communityRequest = CommunityMembershipRequestDto(
+        id: "",
+        publicKey: "",
+        chatId: "",
+        communityId: communityId,
+        state: int(RequestToJoinType.Loading),
+        our: ""
+      )
+      self.myCommunityRequests.add(communityRequest)
+      self.events.emit(SIGNAL_COMMUNITY_MY_REQUEST_ADDED, CommunityRequestArgs(communityRequest: communityRequest))
```

Comment:
So this part looks a bit backwards to me. Seems like we're optimistically updating our locally known requests.

Does anything speak against:

1. Make request async
2. Wait for callback when request was made (just like we do with waiting for synchronous call to be done)
3. When callback doesn't contain an error, we know it was successful, **then** emit ADDED event
4. In case of error, emit error event so we can show a notification accordingly